src/jwplayer/players/shared/JwPlayerWrapperWithStrategyRules.tsx

```diff
@@ -0,0 +1,116 @@
+import useBeforeJwpWrapperRendered from 'jwplayer/utils/useBeforeJwpWrapperRendered';
+import useScript from 'jwplayer/utils/useScript';
+
+interface WindowJWPlayer extends Window {
+	jwplayer?: JWPlayerApi;
+	jwplacements?: JWPlacementApi;
+	jwDataStore?: Record<string, unknown>;
+	sponsoredVideos?: string[];
+}
+
+declare let window: WindowJWPlayer;
+
+const JwPlayerWrapperWithStrategyRules: React.FC<JwPlayerWrapperProps> = ({
+	config,
+	onReady,
+	getDismissed = () => false,
+	onComplete,
+	className,
+	shouldLoadSponsoredContentList = true,
+	jwPlayerContainerEmbedId = 'featured-video__player',
+	vastUrl,
+	parentClassName,
+}) => {
+	const strategyRulesPlacementId = '21rL5wJF';
+	const recommendationPlaylistId = 'FOhaD53w';
+
+	const registerEventHandlers = (playerInstance) => {
+		playerInstance.on(JWEvents.AD_PAUSE, ({ pauseReason, viewable }: JWPauseEvent) => {
+			// Keep playing the ad when the user closed the mini player
+			if (viewable === 0 && pauseReason === 'external') {
+				playerInstance.play();
+			}
+		});
+
+		playerInstance.on(JWEvents.PLAY, ({ playReason, viewable }: JWPlayEvent) => {
+			const dismissed = getDismissed();
+			// Pause the content play when the user closed the mini player playing the ad
+			if (dismissed && viewable === 0 && playReason === 'autostart') {
+				playerInstance.pause();
+			}
+		});
+
+		playerInstance.on(JWEvents.READY, () => {
+			recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_READY);
+			triggerVideoMetric('ready');
+			// only add the events after the player is ready
+			jwPlayerPlaybackTracker({ event_name: 'video_player_ready' });
+			addBaseTrackingEvents(playerInstance);
+
+			if (onReady) {
+				console.debug('Player ready...');
+				onReady(playerInstance);
+			}
+		});
+
+		if (typeof onComplete === 'function') {
+			playerInstance.on(JWEvents.COMPLETE, () => onComplete());
+		}
+	};
+
+	const jwPlayerLoaded = (payload: JWPPlacementReadyResponse) => {
+		console.debug('Placement Embed Complete: ', payload.placementId, payload.playerDivId, payload.player);
+
+		// Set the max_resolution param for related videos
+		if (typeof window?.jwplayer?.defaults?.related?.file === 'string') {
+			window.jwplayer.defaults.related.file = window.jwplayer.defaults.related.file + '&max_resolution=1280';
+		}
+
+		jwPlayerPlaybackTracker({ event_name: 'video_player_load' });
+		recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_SCRIPTS_LOAD_READY);
+		triggerVideoMetric('loaded');
+
+		setConfig(config);
+
+		const playerInstance = payload.player;
+		registerEventHandlers(playerInstance);
+		setPlayer(playerInstance);
+	};
+
+	const initPlayer = () => {
+		console.debug('Strategy rules enabled: the embed will be loaded inline', vastUrl);
+	};
+
+	const { setPlayer, setConfig } = useContext(PlayerContext);
+	useBeforeJwpWrapperRendered(initPlayer, shouldLoadSponsoredContentList);
+
+	const selector = `.${parentClassName} .strategyRulesWrapper`;
```

Comment:
We should rather use react's ref to pass the element reference instead of using class selectors.